crates/walrus-service/src/node.rs

```diff
@@ -7837,6 +7992,107 @@ mod tests {
             Ok(())
         }
 
+        #[walrus_simtest]
+        async fn flush_after_registration() -> TestResult {
+            walrus_test_utils::init_tracing();
+
+            let flush_blocked = Arc::new(Notify::new());
+            let release_flush = Arc::new(Notify::new());
+            let flush_blocked_clone = flush_blocked.clone();
+            let release_flush_clone = release_flush.clone();
+            let certify_started = Arc::new(Notify::new());
+            let certify_started_clone = certify_started.clone();
+
+            register_fail_point_async("fail_point_flush_pending_caches_with_logging", move || {
+                let flush_blocked = flush_blocked_clone.clone();
+                let release_flush = release_flush_clone.clone();
+                async move {
+                    flush_blocked.notify_one();
+                    release_flush.notified().await;
+                }
+            });
+
+            register_fail_point_async("fail_point_process_blob_certified_event", move || {
+                let certify_started = certify_started_clone.clone();
+                async move {
+                    certify_started.notify_one();
+                }
+            });
+
+            let (cluster, events) = cluster_at_epoch1_without_blobs(&[&[0, 1, 2, 3]], None).await?;
+            let storage_node = cluster.nodes[0].storage_node.clone();
+
+            let encoded = EncodedBlob::new(BLOB, cluster.encoding_config());
+            let blob_id = *encoded.blob_id();
+            let sliver_pair = encoded.assigned_sliver_pair(SHARD_INDEX);
+
+            storage_node
+                .store_metadata(
+                    encoded.metadata.clone().into_unverified(),
+                    UploadIntent::Pending,
+                )
+                .await?;
+            storage_node
+                .store_sliver(
+                    blob_id,
+                    sliver_pair.index(),
+                    Sliver::Primary(sliver_pair.primary.clone()),
+                    UploadIntent::Pending,
+                )
+                .await?;
+
+            events.send(BlobRegistered::for_testing(blob_id).into())?;
+            events.send(BlobCertified::for_testing(blob_id).into())?;
+
+            tokio::time::timeout(Duration::from_secs(5), flush_blocked.notified())
+                .await
+                .expect("cache flush should hit failpoint before certification runs");
+            tokio::time::timeout(Duration::from_millis(100), certify_started.notified())
```

Comment:
nit: 100ms seem short. Given the test uses simulated time, maybe increase this to 1s?